meta.go

```diff
@@ -74,3 +74,24 @@ func (m Meta) Clone() Meta {
 	m.Encoder = m.Encoder.Clone()
 	return m
 }
+
+// Enabled returns true if logging a message at a particular level is enabled.
+func (m Meta) Enabled(lvl Level, msg string) bool {
+	return lvl >= m.Level()
+}
+
+// Check returns a CheckedMessage logging the given message is Enabled, nil
+// otherwise.
+func (m Meta) Check(log Logger, lvl Level, msg string) *CheckedMessage {
+	switch lvl {
+	case PanicLevel, FatalLevel:
+		// Panic and Fatal should always cause a panic/exit, even if the level
+		// is disabled.
+		break
+	default:
+		if !m.Enabled(lvl, msg) {
+			return nil
+		}
+	}
+	return NewCheckedMessage(log, lvl, msg)
```

Comment:
Unless there's another use case for it, I'm all for un-exporting `NewCheckedMessage`. Existing callers should all have access to a `Meta`, which they can use instead.